clojure-ts-mode.el

```diff
@@ -2048,6 +2048,76 @@ value is `clojure-ts-thread-all-but-last'."
       (user-error "No string or keyword at point")))
     (goto-char pos)))
 
+(defun clojure-ts--collection-node-at-point ()
+  "Return node at point that represent a collection."
+  (when-let* ((node (thread-first (point)
+                                  (treesit-node-at 'clojure)
+                                  (treesit-parent-until (rx bol
+                                                            (or "map_lit"
+                                                                "vec_lit"
+                                                                "set_lit"
+                                                                "list_lit"
+                                                                "quoting_lit")
+                                                            eol)))))
+    (cond
+     ;; If node is a list, check if it's quoted.
+     ((string= (treesit-node-type node) "list_lit")
+      (if-let* ((parent (treesit-node-parent node))
+                ((string= (treesit-node-type parent) "quoting_lit")))
+          parent
+        node))
+     ;; If the point is at the quote character, check if the child node is a
+     ;; list.
+     ((string= (treesit-node-type node) "quoting_lit")
+      (when-let* ((first-child (clojure-ts--node-child-skip-metadata node 0))
+                  ((string= (treesit-node-type first-child) "list_lit")))
+        node))
+     (t node))))
+
+(defun clojure-ts--convert-collection (coll-open coll-close &optional prefix)
```

Comment:
I'd probably name `coll-open` and `coll-close` something `delim-open` and `delim-close`. Perhaps those can also be a single `coll-delims` param that's a pair or a alist/plist.

Reply:
Actually we can probably pass only opening paren and use `(matching-paren delim-open)` to get the pair. WDYT?